docs: add english document (#17)
ecomfe/veui-docs: one/docs/en-US/components/anchor.md

# Anchor

> ## Demo

Reviewer: ```suggestion
## Demos
```

> ### Basic Anchor

Reviewer: ```suggestion
### Basic anchor
```

> Use [`container`](#props-container) property to set the container in which scrolling occurs. When the `Anchor` component is not in the container (that is, it does not scroll with the container), you do not need to set [`sticky`](#props-sticky ) property.

Reviewer: ```suggestion
Use the [`container`](#props-container) prop to set the container where scrolling occurs. When the `Anchor` component is not in the container (i.e. it does not scroll with the container), there's no need to set the [`sticky`](#props-sticky ) prop.
```

[[ demo src="/demo/anchor/normal.vue" ]]

> ### Adsorption Anchor

Reviewer: ```suggestion
### Sticky anchor
```

> When `Anchor` component is in the container (rolling with the container), you can set [`sticky`](#props-sticky) property to control the adsorption of the component when it rolls out of the container.

Reviewer: ```suggestion
When the `Anchor` component is in the container (rolling with the container), the [`sticky`](#props-sticky) prop can be set to control the attachment of the component when it scrolls out of the container.
```

[[ demo src="/demo/anchor/sticky.vue" ]]

### Specify Offset

> Use [`target-offset`](#props-target-offset) property to control where the anchor point scrolls to the container to start to be active.

Reviewer: ```suggestion
Use the [`target-offset`](#props-target-offset) prop to control where the anchor scrolls into the container and becomes active.
```

> Use [`sticky-offset`](#props-sticky-offset) property to control where the `Anchor` is in the container to start the adsorption.

Reviewer: ```suggestion
Use the [`sticky-offset`](#props-sticky-offset) prop to control where in the container the `Anchor` starts to attach.
```

[[ demo src="/demo/anchor/offset.vue" ]]

## API

### Props

| Name | Types | Default | Description |
| -- | -- | -- | -- |
| ``items`` | `Array<{value, label, children}>` | `[]` | [^items] |

> | ``sticky`` | `boolean=` | `false` | Whether produce adsorption effect. |

Reviewer: ```suggestion
| ``sticky`` | `boolean=` | `false` | Whether to apply sticky effect. |
```

> | ``container`` | `string | HTMLElement | Window=` | - | `Anchor` adsorption and determine the container referenced by the anchor point activation. |

Reviewer: ```suggestion
| ``container`` | `string | HTMLElement | Window=` | - | The container which the `Anchor` component attaches to and checks active state against. |
```

> | ``target-offset`` | `string | number=` | `0` | When an anchor point is in the `target-offset` position to the container, then the corresponding anchor point link is active. The numeric type value is a `px`, and a string such as `'10%'` can also be passed in. The calculation will be based on the proportion of the height of the container corresponding to the `container`. |

Reviewer: ```suggestion
| ``target-offset`` | `string | number=` | `0` | When an anchor is at the `target-offset` position of the container, then the corresponding anchor link is active. The numeric type value is a `px` value, or a percentage string such as `'10%'` can also be used, which will be calculated based on the height of the `container`. |
```

> | ``sticky-offset`` | `string | number=` | `0` | sticky `Anchor`, when the container rolls to the position of `sticky-offset`, then the `Anchor` starts to adsorb. Different value types have the same meaning as [`target-offset`](#props-target-offset) property. |

Reviewer: ```suggestion
| ``sticky-offset`` | `string | number=` | `0` | For a sticky `Anchor`, when the container scrolls to the `sticky-offset` position, then the `Anchor` starts to attach. Different value types have the same meaning as the [`target-offset`](#props-target-offset) prop. |
```

^^^items

> Data source array, item type is `{label, value, children, ...}`.

Reviewer: ```suggestion
The datasource array. The type of each item is `{label, value, children, ...}`.
```

> +++Field

Reviewer: ```suggestion
+++Properties
```

| Name | Type | Description |
| -- | -- | -- |

> | `label` | `string` | Text description of the node. |

Reviewer: ```suggestion
| `label` | `string` | The descriptive labe of the item. |
```

> | `value` | `string` | Value corresponding to the node is generally an in-page hash, such as `#button`. |

Reviewer: ```suggestion
| `value` | `string` | The value of the item, usually an in-page hash, such as `#button`. |
```

> | `children` | `Array<Object>=` | Child node array of the node, Array item type is the same as the [`items`](#props-items) array item. |

Reviewer: ```suggestion
| `children` | `Array<Object>=` | The children array of the item, with the same type as the [`items`](#props-items) array item. |
```

+++
^^^

### Slots

| Name | Description |
| -- | -- |
| ``item`` | [^slot-item] |
| ``item-label`` | [^slot-item-label] |

^^^slot-item

> Render each anchor link.

Reviewer: ```suggestion
Renders each anchor link.
```

Default content: anchor link.

> Scope parameters, see [`items`](#props-items) property details.

Reviewer: ```suggestion
See the [`items`](#props-items) prop for scope properties details.
```

^^^

^^^slot-item-label

> Render the text of each anchor link.

Reviewer: ```suggestion
Renders the label of each anchor link.
```

> Default content: the description text `label` of the anchor link.

Reviewer: ```suggestion
Default content: the `label` of the anchor link.
```

> Scope parameters, see [`item`](#slots-item) slots.

Reviewer: ```suggestion
See the [`items`](#props-items) prop for scope properties details.
```
